### ISSUE TITLE
Missing query_root when no permissions are set

### ISSUE THREAD

**Issue by FredyC** [hasura/graphql-engine#4460]
My use is simply based on having a single mutation for unauthenticated users which basically is an action to signUp/signIn. There is no access to any data.

When running an introspection query in such a case, the `query_root` does not exist at all which is trouble for some external tools.

By spec, the query_root should be always defined even if empty.

http://spec.graphql.org/draft/#sec-Root-Operation-Types

![image](https://user-images.githubusercontent.com/1096340/79603696-1d930400-80ed-11ea-9fc2-cb34497a749a.png)

**Comment by ngocdiep** [comment]
I set ENV for example:
`HASURA_GRAPHQL_UNAUTHORIZED_ROLE: "anonymous"`
then set permission to allow "select", I can query data as an anonymous user

**Comment by nicholasengleman** [comment]
Yeah could we at least get a more descriptive error message when no permissions are set?

**Comment by FredyC** [comment]
Let's not pursue workarounds here. There are some available so it's not a critical issue, but this should be fixed to fulfill spec.

**Comment by madmaxlax** [comment]
i believe I am running into this as well, 
we are trying to use the codegen tool mentioned here, https://hasura.io/learn/graphql/typescript-react-apollo/codegen/
we are still setting up our auth checking, so for the code gen i am just using x-hasura-admin-secret, 

and I am getting `cannot query field "something_definitely_there" on type "Query"` errors for fields that have permissions 

to help debug the issue i tried https://github.com/apollographql/apollo-tooling#apollo-clientdownload-schema-output
and interestingly i see all the fields i'd expect under `type query_root {...all_my_fields...}` but not under `type Query {...none...}` which is maybe the issue ?

note: this seems to be an issue on the code-gen side as well: 
https://github.com/dotansimha/graphql-code-generator/issues/3982 
after version 1.15.0 it is incorrectly goring the SchemaDefinition, and just assuming the typename is Query
```
schema {
  query: query_root
  mutation: mutation_root
  subscription: subscription_root
}
```

but could help if hasura maybe put all the same fields for the schema in Query as query_root?
